lib/src/uri_pattern.dart

```diff
@@ -45,9 +42,20 @@ class UriMatch {
       other is UriMatch &&
       pattern == other.pattern &&
       input == other.input &&
-      mapsEqual(parameters, other.parameters) &&
+      parameters.equals(other.parameters) &&
       rest == other.rest;
 
   @override
-  int get hashCode => hash4(pattern, input, parameters.toString(), rest);
+  int get hashCode => Object.hash(pattern, input, parameters.toString(), rest);
+}
+
+extension on Map<String, String?> {
+  bool equals(Map<String, String?> other) {
```

Comment:
can we just use collectionEquals from dart:collection?

Reply:
We would have to add a dependency on package:collection to get that afaik, and we only need shallow equality which is trivial to implement here.